lib/deimos/active_record_consume/batch_consumption.rb

```diff
@@ -145,15 +171,27 @@ def deleted_query(records)
       # Get the set of attribute names that uniquely identify messages in the
       # batch. Requires at least one record.
       # @param records [Array<Message>] Non-empty list of messages.
+      # @param _klass [ActiveRecord::Class] Class Name can be used to fetch columns
       # @return [Array<String>] List of attribute names.
       # @raise If records is empty.
-      def key_columns(records)
+      def key_columns(records, _klass)
```

Comment:
I think we can change this method so that it works in both cases and still allows `super`:

```ruby
def key_columns(records, klass)
  if klass.nil? # check records
    raise 'Cannot determine key from empty batch' if records.empty?

        first_key = records.first.key
        record_key(first_key).keys
  else
     raise 'Must implement key_columns method for associations!
  end
end
```

In this case:
* Calling it with no class passed in (default case) will do the current default behavior. You can call `super` here to get the current default behavior.
* Calling it with a subclass will have no default behavior and will raise an error. You can't call `super` here anyway because we don't have any defined behavior.

Reply:
Done.